cmake/oidn_metal.cmake

```diff
@@ -17,6 +17,13 @@ function(metallib_target_add_sources target metallib)
   foreach(def ${METAL_COMPILE_DEFINITIONS})
     list(APPEND compile_defs "-D${def}")
   endforeach()
+  
+  # Select the appropriate SDK based on OIDN_METAL_IOS
+    if(OIDN_METAL_IOS)
+        set(sdk "iphoneos")
+    else()
+        set(sdk "macosx")
+    endif()
```

Comment:
It would be nice to add support for iPadOS too.

Reply:
Great point, they do share the same sdk. I'll look at building and validating for iPadOS. On that note, the libraries should be the same for xros so that may be worth exploring too.

Reply:
Confirmed a working build on iPadOS, added `visionOS` to `CMAKE_SYSTEM_NAME` but I will be unable to confirm a build works as I don't have the hardware. In the meantime, it may be beneficial to set up flags for the iphonesimulator which should make it possible for the library to be used in previews and simulators for ios/ipados.

Reply:
I think compiling for simulators and investigating xros might be scope creep for this PR, I will investigate in the future.